### DISCUSSION TITLE
API returns 405

### DISCUSSION THREAD

**Discussion by islamsamy214** [mailcarrierapp/mailcarrier#13] [General]
dude, I just tried it with docker and it gives me 405 NOT ALLOWED

**Comment by danilopolani** [comment]
Can you past here the cURL of the request you're sending?

**Reply by islamsamy214** [reply]
I used postman for that "http://localhost:8080/api/send" , I created a User then I tried to return an authorization key for the login to send a mail, but I found there wasn't any api for the login, but this wasn't the problem I just brought the "send" out from the api middleware just to try it whether it is workin or not, although I declared the sender in the api just make sure it not using the Loggedin user, but 405 was result in both, and this was my raw data: {
    "recipient":"foo@example.com",
    "subject":"My first email",
    "template":"test",
    "sender": {
        "email": "foo@example.com",
        "name": "foo"
    }
}, and of course I made sure that it was JSON type

**Reply by islamsamy214** [reply]
I just created a token from php artisan mailcarrier:token and used, it is not working too

**Reply by danilopolani** [reply]
The 405 is `Method Not Allowed`, so not related to authentication but to the HTTP Verb. It might sounds weird, but are 100% sure you're sending a `POST` request to that endpoint?

**Reply by islamsamy214** [reply]
Oops, yes I think I forgot that, but now went to taking some time then it is timing out with 504

**Reply by danilopolani** [reply]
Did you configure your SMTP provider? By default it does use nothing to send emails, you have to pass your config or with a `.env` file and then passing it with `--env-file`, or the single environment variables:

**With env file**
```env 
MAIL_MAILER=smtp
MAIL_HOST=sandbox.smtp.mailtrap.io
MAIL_PORT=2525
MAIL_USERNAME=<mailtrap_setting_username>
MAIL_PASSWORD=<mailtrap_setting_password>
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS="no-reply@example.org"
MAIL_FROM_NAME="MyApp"
``` 

```shell
docker run -p 8080:80 --env-file .env mailcarrier/mailcarrier
``` 

Or env-by-env

```shell
docker run \
  -p 8080:80 \
  -e MAIL_MAILER=smtp
  -e MAIL_HOST=sandbox.smtp.mailtrap.io
  ... and so on
  mailcarrier/mailcarrier
```

**Reply by islamsamy214** [reply]
Ooh dude, I thought this app would stand as the SMTP, that is why it got my interest, but if so what is benefit if would end up with an SMTP!, but yeah great work thank you 🙏

**Reply by danilopolani** [reply]
Yeah sorry, probably I should specify it in the docs too 😄 Glad to hear you solved tho!
